File: tools/sap/src/main/java/top/bettercode/summer/tools/sap/connection/SapException.java
package top.bettercode.summer.tools.sap.connection;

/**
 * @author Peter Wu
 */
public class SapException extends RuntimeException {

  private static final long serialVersionUID = 1L;


  public SapException(String message) {
    super("SAP系统：" + message);
  }

  public SapException(Throwable cause) {
    super("SAP系统：" + cause.getMessage(), cause);
  }

  public SapException(String message, Throwable cause) {
    super("SAP系统：" + message, cause);
  }

}

File: tools/sap/src/main/kotlin/top/bettercode/summer/tools/sap/connection/SapException.kt
package top.bettercode.summer.tools.sap.connection

/**
 * @author Peter Wu
 */
class SapException : RuntimeException {
    constructor(message: String) : super("SAP系统：$message")
    constructor(cause: Throwable) : super(if (cause is SapException || cause is SapSysException) cause.message else "SAP系统：" + cause.message, cause)
    constructor(message: String?, cause: Throwable?) : super("SAP系统：$message", cause)

    companion object {
        private const val serialVersionUID = 1L
    }
}
